build_with_cmake/node-addon-api/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.9)
+cmake_policy(SET CMP0042 NEW)
+set (CMAKE_CXX_STANDARD 11)
+
+project (build-node-addon-api-with-cmake)
+include_directories(${CMAKE_JS_INC})
+file(GLOB SOURCE_FILES "hello.cc")
+add_library(${PROJECT_NAME} SHARED ${SOURCE_FILES} ${CMAKE_JS_SRC})
+set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "" SUFFIX ".node")
+target_link_libraries(${PROJECT_NAME} ${CMAKE_JS_LIB})
+
+# define NPI_VERSION
+add_definitions(-DNAPI_VERSION=2)
+
+# Include N-API wrappers
+target_include_directories(${PROJECT_NAME} PRIVATE "${CMAKE_SOURCE_DIR}/node_modules/node-addon-api")
```

Comment:
Could be possible call:
`node -p "require('node-addon-api').include" ` that return the absolute path for your include dir  instead to set it by hand?

Reply:
Good find, Nicola. 

I've updated `CMakeLists.txt` to use the technique recommended by the [CMake.js team](https://github.com/cmake-js/cmake-js#n-api-and-node-addon-api).

This technique was failing for me before, but now it's working again. It's probably due to the three lines I added earlier to the very top of `CMakeLists.txt `.